document/src/vespa/document/base/field.h

```diff
@@ -28,16 +28,35 @@ class Field final : public vespalib::FieldBase,
     const DataType *_dataType;
     int             _fieldId;
 public:
-    typedef std::shared_ptr<const Field> CSP;
-    typedef std::shared_ptr<Field> SP;
+    using CSP = std::shared_ptr<const Field>;
+    using SP = std::shared_ptr<Field>;
+    using CPtr = const Field *;
 
     struct FieldPtrComparator {
-        bool operator()(const Field* f1, const Field* f2) const {
+        bool operator()(CPtr f1, CPtr f2) const {
             return (*f1 < *f2);
         }
     };
 
-    using Set = std::set<const Field*, FieldPtrComparator>;
+    class Set {
+    public:
+        class Builder {
+        public:
+            Builder & reserve(size_t sz) { _vector.reserve(sz); return *this; }
+            Builder & insert(CPtr field) { _vector.emplace_back(field); return *this; }
```

Comment:
I would prefer 'add' over 'insert' here, since it implies less. Also, I would prefer 'push_back' over 'emplace_back' for values that already have the appropriate type.

Reply:
Fixed